pyqtgraph/functions.py

```diff
@@ -1870,6 +1870,162 @@ def downsample(data, n, axis=0, xvals='subsample'):
         return MetaArray(d2, info=info)
 
 
+def _compute_backfill_indices(isfinite):
+    # the presence of inf/nans result in an empty QPainterPath being generated
+    # this behavior started in Qt 5.12.3 and was introduced in this commit
+    # https://github.com/qt/qtbase/commit/c04bd30de072793faee5166cff866a4c4e0a9dd7
+    # We therefore replace non-finite values
+
+    # credit: Divakar https://stackoverflow.com/a/41191127/643629
+    mask = ~isfinite
+    idx = np.arange(len(isfinite))
+    idx[mask] = -1
+    np.maximum.accumulate(idx, out=idx)
+    first = np.searchsorted(idx, 0)
+    if first < len(isfinite):
+        # Replace all non-finite entries from beginning of arr with the first finite one
+        idx[:first] = first
+        return idx
+    else:
+        return None
+
+
+def _arrayToQPath_all(x, y, finiteCheck):
+    n = x.shape[0]
+    if n == 0:
+        return QtGui.QPainterPath()
+
+    if finiteCheck:
+        isfinite = np.isfinite(x) & np.isfinite(y)
+        all_isfinite = np.all(isfinite)
+
+    chunksize = 10000
+    numchunks = (n + chunksize - 1) // chunksize
+    minchunks = 3
+
+    if numchunks < minchunks:
+        # too few chunks, batching would be a pessimization
+        poly = create_qpolygonf(n)
+        arr = ndarray_from_qpolygonf(poly)
+
+        backfill_idx = None
+        if finiteCheck and not all_isfinite:
+            backfill_idx = _compute_backfill_indices(isfinite)
+
+        if backfill_idx is None:
+            arr[:, 0] = x
+            arr[:, 1] = y
+        else:
+            arr[:, 0] = x[backfill_idx]
+            arr[:, 1] = y[backfill_idx]
+
+        path = QtGui.QPainterPath()
+        if hasattr(path, 'reserve'):    # Qt 5.13
+            path.reserve(n)
+        path.addPolygon(poly)
+        return path
+
+    backfill_idx = None
```

Comment:
nit-pick: If we move this chunk before the if-statement above, we can get rid of this chunk within the if-statement as well

Reply:
You are right. It looks the way it does now because prior to the 4th commit, the two "if" branches had different code.